### DISCUSSION TITLE
How to use find And count in where clause

### DISCUSSION THREAD

**Discussion by HunterJS-bit** [mikro-orm/mikro-orm#1653] [Q&A]
Hi guys, I have the following case

I have a relation between Authors and BooksAuthor can have many books, and books can have many authors

I have a query like this

`const res = await orm.em.find(Author,[ 'books.author']});`

where I get Authors and their books with additional authors

How would I count relations and use count in where clause for example I would like to count book relations, and count how many are there other authors which name start with 'M' for example and use that in where clause?

**Comment by B4nan** [comment]
For complex queries like that you will need a query builder or custom sql fragments, or maybe subqueries. So first try to craft the query you need yourself, then I will be happy to help with how to instruct the ORM for it.
